flytekit/core/type_engine.py

```diff
@@ -596,6 +596,58 @@ def guess_python_type(self, literal_type: LiteralType) -> Type[list]:
         raise ValueError(f"List transformer cannot reverse {literal_type}")
 
 
+class UnionTransformer(TypeTransformer[T]):
+    """
+    Transformer that handles a univariate typing.Union[T]
+    """
+
+    def __init__(self):
+        super().__init__("Typed Union", typing.Union)
+
+    @staticmethod
+    def get_sub_type(t: Type[T]) -> Type[T]:
```

Comment:
this method does not really make sense since a Union type only ever has multiple "sub types"

i.e. `Union[x]` is by definition equivalent to `x`

I think the way this method is used in `get_literal_type` implies that the type signature should actually be `(t: Type[T]) -> List[Type]` instead

Reply:
will update it, thanks